sdk/communication/Azure.Communication.Identity/tests/SessionRecords/CommunicationIdentityClientLiveTests/CreateUserAndTokenWithChatScope.json

```diff
@@ -0,0 +1,52 @@
+{
+  "Entries": [
+    {
+      "RequestUri": "https://sanitized.communication.azure.com/identities?api-version=2022-10-01",
+      "RequestMethod": "POST",
+      "RequestHeaders": {
+        "Accept": "application/json",
+        "Authorization": "Sanitized",
+        "Content-Length": "34",
+        "Content-Type": "application/json",
+        "traceparent": "00-7b38421f957b7e96fc39d1c4c221a6b4-77abfdf4a7fb4944-00",
+        "User-Agent": "azsdk-net-Communication.Identity/1.3.0-alpha.20230109.1 (.NET 6.0.12; Microsoft Windows 10.0.22621)",
+        "x-ms-client-request-id": "cb37f411de3bcdb5a2da4fd873e74d96",
+        "x-ms-content-sha256": "Sanitized",
+        "x-ms-date": "Mon, 09 Jan 2023 17:55:58 GMT",
+        "x-ms-return-client-request-id": "true"
+      },
+      "RequestBody": {
+        "createTokenWithScopes": [
+          "chat"
+        ]
+      },
+      "StatusCode": 201,
+      "ResponseHeaders": {
+        "api-supported-versions": "2020-07-20-preview2, 2021-02-22-preview1, 2021-03-07, 2021-10-31-preview, 2021-11-01, 2022-06-01, 2022-10-01",
+        "Content-Length": "115",
+        "Content-Type": "application/json; charset=utf-8",
+        "Date": "Mon, 09 Jan 2023 17:56:00 GMT",
+        "MS-CV": "n7bcZi5vpEiJ7dP0RhDxQA.2.0",
+        "Request-Context": "appId=",
+        "Strict-Transport-Security": "max-age=2592000",
+        "X-Azure-Ref": "0MFW8YwAAAAA5ggZlOpFxQ6AH6KP\u002BghWpSEVMMDFFREdFMjAxOQA5ZmM3YjUxOS1hOGNjLTRmODktOTM1ZS1jOTE0OGFlMDllODE=",
+        "X-Cache": "CONFIG_NOCACHE",
+        "x-ms-client-request-id": "cb37f411de3bcdb5a2da4fd873e74d96",
+        "X-Processing-Time": "132ms"
+      },
+      "ResponseBody": {
+        "identity": {
+          "id": "Sanitized"
+        },
+        "accessToken": {
+          "token": "Sanitized",
+          "expiresOn": "2023-01-10T17:56:00.7844833\u002B00:00"
+        }
+      }
+    }
+  ],
+  "Variables": {
+    "COMMUNICATION_LIVETEST_DYNAMIC_CONNECTION_STRING": "endpoint=https://smstestapp.communication.azure.com/;accesskey=Kg==",
```

Comment:
should the URL be also sanitized?

Reply:
Sanitizers only work for a JSON in requests and responses and not to the "Variables" field. Some of the older recordings have this value sanitized and some not. My suggestion is that at some point of time the behavior of sanitizers library changed.
I investigated how the information is processed for sanitization and discovered that a separate Kestrel server is run by a sanitizer library that performs the sanitization. The code for this server is not accessible. None of the additional sanitizers with different syntax that I've tried adding has any influence on the 'Variables' field.
However, considering that some of the old recordings already have this value and some SDKs do no sanitization at all,  this is not a secret value anymore (even if it used to be). So my suggestion is to go with the recordings that are created and discuss further if we need to have an additional story for fixing sanitizers (if this can somehow be done from the code that we own).

@petrsvihlik any ideas on this?

Reply:
It's surprising to say the least. I'd expect everything in the JSON to be sanitizable. And it seems it was the case before...unless someone really has an ACS resource named sanitized.communication.azure.com :)

But it seems @AikoBB managed to get everything sanitized in this PR: https://github.com/Azure/azure-sdk-for-net/commit/fc83708c7ea92408e26d82c514e14111d261c283#diff-8c6c38ddd926f56bc5b531c52f146a83d83f1181d2dd08e7a626fe1c16d588a0 maybe she'll be able to remember and help you

Reply:
@petrsvihlik yes, there were recent changes in the sanitizer code that I noticed when I was debugging but I couldn't find any connection to not sanitizing 'Variables'.

I can manually edit the files and replace that URI with a sanitized value but it'll break the concept of having autogenerated filed. Besides, what's the purpose of sanitizing? I thought that it's to hide sensitive information like tokens and passwords from external users. As the value in question was already exposed in our SDK and a couple of other SDKs, does it make sense to still sanitize it?

Reply:
This very value might have been exposed, however, everyone uses different ACS resources to run the tests and it'd be better to have them sanitized. (We probably don't want to have store JSONs with `my-gummy-bear-service.communication.azure.com` or worse.) Can you pls quickly check with EngSys if the sanitization logic has intentionally changed?

Reply:
I did not remember how I achieved it, but what I remember I did not do anything regarding to sanitizers,  maybe it manually updated. We can reach out to the core team on how we can include those values to be sanitized, I guess there should be a solution like in [java](https://github.com/Azure/azure-sdk-for-java/blob/11d31c5476bd61208de0f501485b38450e64d2dd/sdk/communication/azure-communication-identity/src/test/java/com/azure/communication/identity/CommunicationIdentityClientTestBase.java#L187). As we are sanitizing the resource urls I'd vote for sanitize them everywhere, whether we will do in this PR or create another US for that